vali_objects/utils/p2p_syncer.py

```diff
@@ -350,8 +350,9 @@ def heuristic_resolve_positions(self, position_matrix, positions_threshold, matc
                                 and set([match["position_uuid"] for match in matches]).isdisjoint(matched_position_uuids)):
                             # see if no elements from matches have already appeared in uuid_matched_positions
                             bt.logging.info(f"Miner hotkey {miner_hotkey} has matches {[m['position_uuid'] for m in matches]}")
-                            # bt.logging.info(f"uuid_matched {uuid_matched_positions_set}")
-                            matched_positions.append(matches[0])
+                            num_median_orders = len(matches[len(matches)//2]["orders"])
+                            matches_with_median_num_orders = [match for match in matches if len(match["orders"]) == num_median_orders]
```

Comment:
should sort by position_uuid before taking element 0 so that it is deterministic

Reply:
The matches are already sorted by number of orders and then position_uuid in find_match. This just gets all the matches with the median number of orders, and returns the first element (i.e. the first one sorted by position_uuid)

Reply:
Ah apologies this is good